### DISCUSSION TITLE
CF don't wake up(freeze) after customize firmware with IRQhandler

### DISCUSSION THREAD

**Discussion by kimseong-yeon** [bitcraze/discussions#618] [Q&A]
HI. i'm doing project for implement 2 encoder motor with crazyflie.

first i intend to get **encoder data** using **GPIO interrupt and IRQhandler with GPIO port 1,2,3,4 (PB8,5,4, PC12)**
and i wrote these in custom driver, but at IRQHandler, there are conflicts with **exti.c**

so i  moved these into exti.c and made build and upload with no problem.
but Crazyflie doesn't wake up(at client, connection does not proceed), and CF's 2 blue led is just on with no blinking.

i used TIM12 with DMA for encoder data time. but i'm not sure what is problem.
exti.c is like below
```
void __attribute__((used)) EXTI4_IRQHandler(void)
{
  //added
  uint8_t cur_AB = 0;

  DMA_Cmd(DMA1_Stream5, DISABLE);
  TIM_Cmd(TIM12, DISABLE);

  cur_AB = cur_AB << 1;
  cur_AB |= (GPIOB->IDR & GPIO_Pin_4) >> 3;
  encoderBCount += (cur_AB & 0x03) - 1;

  DMA_Cmd(DMA1_Stream6, ENABLE);
  TIM_Cmd(TIM12, ENABLE);
  //added

  NVIC_ClearPendingIRQ(EXTI4_IRQn);
  EXTI_ClearITPendingBit(EXTI_Line4);
  EXTI4_Callback();
}
```
full code is like .zip
i don't think i write informations properly, if you can see some problems, please let me know.

[custom.zip](https://github.com/bitcraze/discussions/files/10963489/custom.zip)

**Comment by hmllr** [comment]
Hi :)
This is not the topic I know most about - however, I think you cannot use the same interrupt line for two purposes. EXTI4 is already used for uartslkTxenFlowctrlIsr(), which means some uart syslink transmit enable flow control, so if you suddenly map a second pin to this line as well there will be conflicts. You will need to choose a line which is not already in use (like EXTI12 which is on IO4 on the extension headers). Also I would recommend to move your code into the callback function to make your code more portable and to have a better overview.
Hope this helps,
Hanna
